oss-internship-2020/libtiff/example/main_sandboxed.cc

```diff
@@ -0,0 +1,305 @@
+  uint8_t* target = buffer.GetData() + cluster * 6;
+
+  if (!std::memcmp(target, expected_cluster.data(), 6)) {
+    return absl::OkStatus();
+  }
+
+  return absl::InternalError(
+      absl::StrCat("Cluster ", cluster, " did not match expected results.\n",
+                   "Expect: ", expected_cluster[0], "\t", expected_cluster[1],
+                   "\t", expected_cluster[4], "\t", expected_cluster[5], "\t",
+                   expected_cluster[2], "\t", expected_cluster[3], "\n"));
+}
+
+absl::Status CheckRgbPixel(int pixel, int min_red, int max_red, int min_green,
+                           int max_green, int min_blue, int max_blue,
+                           const sapi::v::Array<uint8_t>& buffer) {
+  uint8_t* rgb = buffer.GetData() + 3 * pixel;
+
+  if (rgb[0] >= min_red && rgb[0] <= max_red && rgb[1] >= min_green &&
+      rgb[1] <= max_green && rgb[2] >= min_blue && rgb[2] <= max_blue) {
+    return absl::OkStatus();
+  }
+
+  return absl::InternalError(absl::StrCat(
+      "Pixel ", pixel, " did not match expected results.\n", "Got R=", rgb[0],
+      " (expected ", min_red, "..", max_red, "), G=", rgb[1], " (expected ",
+      min_green, "..", max_green, "), B=", rgb[2], " (expected ", min_blue,
+      "..", max_blue, ")\n"));
+}
+
+absl::Status CheckRgbaPixel(int pixel, int min_red, int max_red, int min_green,
+                            int max_green, int min_blue, int max_blue,
+                            int min_alpha, int max_alpha,
+                            const sapi::v::Array<unsigned>& buffer) {
+  // RGBA images are upside down - adjust for normal ordering
+  int adjusted_pixel = pixel % 128 + (127 - (pixel / 128)) * 128;
+  uint32 rgba = buffer[adjusted_pixel];
+
+  if (TIFFGetR(rgba) >= (uint32)min_red && TIFFGetR(rgba) <= (uint32)max_red &&
+      TIFFGetG(rgba) >= (uint32)min_green &&
+      TIFFGetG(rgba) <= (uint32)max_green &&
+      TIFFGetB(rgba) >= (uint32)min_blue &&
+      TIFFGetB(rgba) <= (uint32)max_blue &&
+      TIFFGetA(rgba) >= (uint32)min_alpha &&
+      TIFFGetA(rgba) <= (uint32)max_alpha) {
+    return absl::OkStatus();
+  }
+
+  return absl::InternalError(absl::StrCat(
+      "Pixel ", pixel, " did not match expected results.\n",
+      "Got R=", TIFFGetR(rgba), " (expected ", min_red, "..", max_red,
+      "), G=", TIFFGetG(rgba), " (expected ", min_green, "..", max_green,
+      "), B=", TIFFGetB(rgba), " (expected ", min_blue, "..", max_blue, "), A=",
+      TIFFGetA(rgba), " (expected ", min_alpha, "..", max_alpha, ")\n"));
+}
+
+}  // namespace
+
+std::string GetFilePath(const std::string& dir, const std::string& filename) {
+  return sandbox2::file::JoinPath(dir, "test", "images", filename);
+}
+
+std::string GetFilePath(const std::string filename) {
+  std::string cwd = sandbox2::file_util::fileops::GetCWD();
+  auto find = cwd.rfind("build");
+
+  std::string project_path;
+  if (find == std::string::npos) {
+    LOG(ERROR) << "Something went wrong: CWD don't contain build dir. "
```

Comment:
Maybe add a note saying "falling back to using current working directory as root dir" or make the error fatal? (`LOG(FATAL)` will kill the program)